### ISSUE TITLE
Some simulated long reads have bad alignments

### ISSUE THREAD

**Issue by cytham** [ndierckx/Sim-it#8]
Hi @ndierckx, 

I have used Sim-it to simulate long reads from hg38 genome without SVs. When I aligned these simulated reads to the hg38 genome, I expected them to map well, but however, found that there were many reads with bad alignments which might be caused by unwanted insertions. I have attached a region of the BAM file that contains some of these reads. I have also attached the config file. I simply used the default settings to simulate the reads. Please let me know why I am getting such badly aligned reads. Thanks.

[Nanopore_Test_Sim-it_HAP1-hg38mainA_chr1-9660000-9690000_SAMFILE.txt](https://github.com/ndierckx/Sim-it/files/8531770/Nanopore_Test_Sim-it_HAP1-hg38mainA_chr1-9660000-9690000_SAMFILE.txt)

[config_Sim-it_edit.txt](https://github.com/ndierckx/Sim-it/files/8531754/config_Sim-it_edit.txt)

**Comment by cytham** [comment]
Hi @ndierckx, any idea on the issue?

**Comment by ndierckx** [comment]
Hi,
Sorry I forgot to look at this. I am running some tests now, will let you know if I find the problem

**Comment by ndierckx** [comment]
Hi,

I can't replicate the problem, maybe it is because of the reference you used, can you send me the link, then I will try to simulate from that one..

**Comment by ndierckx** [comment]
I checked the first 3 reads you send me, they match around 88% with chromosome1, just like you selected in the config file. Orr is it not all the reads. These days Nanopore is more like 93%, so you could increase the value in the config file. Maybe some alignments were bad but that's because the alignment software, the reads seems to be correct

**Comment by cytham** [comment]
I used the primary chromosomes (chr1-chrY) from this reference https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/analysisSet/

Why would the reference genome used be a problem?

Here is the IGV image. As you can see, there are many short supplementary alignments in blue which shouldn't be there.
![image](https://user-images.githubusercontent.com/25361260/169548494-63fa3469-896b-491a-8e87-e340a782e204.png)

Here are the read names of some of these reads:
Test_Sim-it_1137_length=22778_HAP1_split_1
Test_Sim-it_1138_length=37087_HAP1_split_1

**Comment by ndierckx** [comment]
Hi,
Sorry for the late response, I had very busy weeks.. Could you send me those original reads, because I can't see what's wrong from the samfile?

**Comment by cytham** [comment]
@ndierckx 
Attached are the two badly aligned reads in TXT format (GitHub doesn't allow me to attach FASTA or BAM format).

You can see from the SAM file that these reads have multiple supplementary alignments, which is unusual. You can also just align them on UCSC (BLAT) and you'll see that their alignments are uneven and not continuous.

[bad_aligned_reads.txt](https://github.com/ndierckx/Sim-it/files/8899852/bad_aligned_reads.txt)

**Comment by ndierckx** [comment]
Thanks!
There doesn't seem to be any deletions or insertions, but there seem to segments inverted in the reads
This should not happen, so will have a look if I can find the issue in the code...

**Comment by ndierckx** [comment]
Hi,
I did a lot of simulations and didn't found the problem, but I ran it on the version that was on my laptop. I tried today on 1.31 from github and I had the same problem, it seemed a line was removed. I removed that version, and will upload a new improved script tomorrow, because I found another issue with the parallelization. Thanks for pointing this out to me, that version has been online too long...
